XLForm/XL/Descriptors/XLFormRowDescriptor.m

```diff
@@ -135,6 +135,12 @@ -(XLFormBaseCell *)cellForFormController:(XLFormViewController * __unused)formCo
             } else {
                 bundle = [NSBundle bundleForClass:NSClassFromString(cellClass)];
                 cellResource = cellClassString;
+                if ([cellClassString rangeOfString:@"."].location != NSNotFound) {
+                    NSArray *components = [cellClassString componentsSeparatedByString:@"."];
+                    cellResource = [components lastObject];
+                } else {
+                    cellResource = cellClassString;
+                }
```

Comment:
@julienpouget this change doesn't avoid the crash if we don't use `NSClassFromString(<Class>.self)`, right?

Reply:
I think we should be able to use `<Class>.self` instead of `NSClassFromString(<Class>.self)` and avoid a crash at line **136**.

Reply:
1. Yes, you need to register your custom cell using `NSStringFromClass(<Class>.self)` to avoid the crash. This way the string representation of your class will contain the fully qualified name `<ModuleName>.<ClassName>`.

2. I agree with you, it would be better to register the custom cell class (using `<Class>.self` directly in the `cellClassesForRowDescriptorTypes` dictionary instead of it's string representation (`NSStringFromClass(<Class>.self)`). In fact, this will work if you do not use a xib file. But in order to instanciate the cell using your custom xib:
 - the class and xib file should have exactly the same name
 - the class name should be registered as a String (this is a mandatory condition cf. line **125** and I wasn't intending to change that).

Hope this makes it clearer.

Reply:
Would be possible to change it to accept `"Class"`, `NSStringFromClass(<Class>.self)` and `<Class>.self`? Remembering that that crash occurs just on iOS 11. In any other iOS version using `<Class>.self` works fine.

Reply:
I think what you are requesting is a different feature and not related to this issue. If you pass `<Class>.self` then the cell will be instantiated from that class using `initWithStyle: reuseIdentifier:` and not loaded from a nib file.